Character.cpp

```diff
@@ -98,20 +100,49 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+
+        // restoring stats to initial values if their initial value is lower than the current values
+        if (hitPoints < *initialHitPoints)
+        {
+            hitPoints = *initialHitPoints;
+        }
+        
+        if (armor < *initialArmorLevel)
+        {
+            armor = *initialArmorLevel;
+        }
+
+        if (attackDamage < *initialAttackDamage)
+        {
+            attackDamage = *initialAttackDamage;
+        }
+
+        // stats get boosted by 10%
+        double hitPointsCopy = hitPoints * 1.10;
+        double armorCopy = armor * 1.10;
+        double attackDamageCopy = attackDamage * 1.10;
+
+        hitPoints = hitPointsCopy;
+        armor = armorCopy;
+        attackDamage = attackDamageCopy;
+
+        *initialHitPoints = hitPoints;
+        *initialArmorLevel = armor;
+        *initialAttackDamage = attackDamage;
```

Comment:
You are doing the same thing with 2 pairs of variables, 3 times over.
Identify the pattern, create a free function that implements the pattern, and call it 3 times with each pair of variables.
You do not need the temp variable, either.
You are also dereferencing the same variables 3 times.  Try to eliminate that, so you only dereference once.
- [ ] revise